test/Microsoft.IdentityModel.Tokens.Tests/AsymmetricSignatureTests.cs

```diff
@@ -51,6 +59,49 @@ public void UnsupportedRSATypes()
             TestUtilities.AssertFailIfErrors(context);
         }
 
+        [Theory, MemberData(nameof(RSASignaturePaddingTheoryData))]
+        public void TestSupportedSignaturePaddingValues(AsymmetricAdapterTheoryData theoryData)
+        {
+            var context = new CompareContext("TestSupportedSignaturePaddingValues");
+            TestUtilities.WriteHeader($"{this}.TestSupportedSignaturePaddingValues");
+
+            try
+            {
+                AppContext.SetSwitch("Switch.Microsoft.IdentityModel.EnablePkcs1SignaturePaddingSwitch", theoryData.EnablePkcs1SignaturePaddingSwitch);
+                var asymmetricAdapter = new AsymmetricAdapter(new RsaSecurityKey(new DerivedRsa(2048)), theoryData.Algorithm, false);
+            }
+            catch(Exception ex)
+            {
+                theoryData.ExpectedException.ProcessException(ex, context);
+            }
+
```

Comment:
Could consider re-setting the state back to default before asserting, funny things happen when using context switches